### DISCUSSION TITLE
one reconstructed example using own dataset(with black ghost around the recostructed people)

### DISCUSSION THREAD

**Discussion by neilgogogo** [NVlabs/instant-ngp#246] [General]
![企业微信截图_16455886629637](https://user-images.githubusercontent.com/53148705/155259486-0f71afe7-c734-4ee8-8177-eaeca42abf40.png)

I tried to input the cmu panoptic data into the data loader and find it is quite wired that there are several reconstructed humans in the scene, meanwhile, the human is lying instead of standing. I guess it might be the reason that the scale of T in extri is not correct? Has anyone encountered this problem?

Many thanks

**Comment by neilgogogo** [comment]
I find in my dataset, the camera coordinate is y right x up, while the fox dataset is y down, so maybe I need to rotate the matrix to y down and see what will happen
![企业微信截图_16456095922051](https://user-images.githubusercontent.com/53148705/155295095-5b727e65-c455-45a4-90de-dc5f14cf6db9.png)

**Comment by neilgogogo** [comment]
https://user-images.githubusercontent.com/53148705/155487657-6f89b59d-1862-469a-89a7-e5be5c0fc911.mp4

After the rotation of the camera coordinate system, the camera's extrinsic parameters seem to be OK, and the reconstructed object converges to one single object. but there is a lot of black fog in the reconstruction results. My training data is a picture with a black background.  When I zoom into the box, I could see the reconstructed human.

**Comment by neilgogogo** [comment]
https://user-images.githubusercontent.com/53148705/155488975-966fcd5e-8ba8-4f4e-984f-346f48c8265e.mp4

here is the zoom-in result, then I find I could filter out the black fog by using  cropping AABB max-min.  Although there are still some ghost which might be the reason of poor extri and intri parameters.

**Comment by qhdqhd** [comment]
How did you calibrate the cameras? using colmap or grundtruth? As far as I know, CMU data have different intrinsic parameters for different cameras, which is not suitable for NeRF.

**Reply by neilgogogo** [reply]
Hi, you need to edit the nerf_dataloader.cu and nerf_dataloader.h, to load different fx, fy, cx, cy into the instant-ngp. In the code, the fx, fy is a vector, but the principle-point x,y is a float value, which means you need to use a vector instead. 

Cheers

**Reply by qhdqhd** [reply]
Thank you.

**Reply by qhdqhd** [reply]
By the way,  You said "filter out the black fog by using cropping AABB max-min", can you explain it ? I meet the same problem.

**Reply by neilgogogo** [reply]
I used the original image later, without masking out the background. And I think the result is better than using the masked image, which has more black area.

But you could also use the AABB max-min instead to filter out the unwanted black ghost, it is just like changing the rendering box and everything outside the box will not be rendered.
![image](https://user-images.githubusercontent.com/53148705/156738774-4889de12-f24e-4819-890e-371e8c814bb1.png)

**Reply by qhdqhd** [reply]
I get it! Thank you. Generally, masking out the background really works.

**Reply by qhdqhd** [reply]
CMU data has 31 HD cameras, which is not enough for NeRF. So the black fog may relate to this?

**Reply by neilgogogo** [reply]
Yes, that might be the reason. I remember in the paper, it said 50-150 training views for Nerf.
